lib/modules/translation/screens/translation_editor.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../widgets/asset.dart';
+
+typedef AssetOverwrite = MapEntry<String, String>;
+
+class TranslationEditorScreen extends StatefulWidget {
+  const TranslationEditorScreen({
+    required this.info,
+    required this.narrative,
+    required this.base,
+    required this.target,
+    Key? key,
+  }) : super(key: key);
+
+  final StoryInfo info;
+  final Story narrative;
+  final Translation base;
+  final Translation target;
+
+  @override
+  State<TranslationEditorScreen> createState() => TranslationEditorState();
+}
+
+class TranslationEditorState extends State<TranslationEditorScreen> {
+  StoryInfo get info => widget.info;
+  Story get narrative => widget.narrative;
+  Translation get base => widget.base;
+  Translation get target => widget.target;
+
+  final changes = <String, AssetOverwrite>{};
+
+  @override
+  void setState(void Function() fn) {
+    super.setState(fn);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Provider.value(
+      value: this,
+      child: Builder(
+        builder: (context) {
+          return WillPopScope(
+            onWillPop: () => showDangerDialog(
+              context,
+              'Leave editor? Unsaved progress will be lost!',
+              confirmText: 'Exit',
+              rejectText: 'Stay',
+            ),
+            child: Scaffold(
+              appBar: AppBar(
+                leading: const RoundedBackButton(),
+                title: const Text('Translations editor'),
+              ),
+              floatingActionButton: changes.isNotEmpty &&
+                      FirebaseAuth.instance.currentUser?.uid ==
+                          info.translationAuthorID
+                  ? FloatingActionButton(
+                      onPressed: () async {
+                        await showLoadingDialog(
+                          context,
+                          applyAssetChanges(info, changes),
+                        );
+                        Navigator.pop(context);
+                      },
+                      child: const Icon(Icons.upload_rounded),
+                    )
+                  : null,
+              body: ListView(
+                children: [
+                  ListTile(
+                    leading: const Icon(Icons.language_rounded),
+                    title: Text(target.language),
+                  ),
+                  buildExplanationTile(context, 'General info'),
+                  // ignore: prefer_const_constructors
+                  Asset('title', icon: Icons.title_rounded),
+                  // ignore: prefer_const_constructors
+                  Asset('description', icon: Icons.description_rounded),
+                  // ignore: prefer_const_constructors
+                  Asset('tags', icon: Icons.tag_rounded),
+                  buildExplanationTile(context, 'Actors'),
```

Comment:
"Characters" is probably a more appropriate word. I think "Actor" is more common when designing games, not stories

Reply:
I agree in general, but we already use this term across the UIs and classes. And we have the same issue with referencing stories as narratives and vice-versa. Overall many inconsistencies. I think now for some time to make a general clean up where we align all the terminology in the app. Maybe after this or the next PR it'll be a good moment for that.

Reply:
Yes. I suggest opening an issue so we don't forget and resolving this comment after